Library/Homebrew/service.rb

```diff
@@ -0,0 +1,99 @@
+# typed: true
+# frozen_string_literal: true
+
+# The {Service} class implements the DSL methods used in a formula's
+# `service` block and stores related instance variables. Most of these methods
+# also return the related instance variable when no argument is provided.
+class Service
+  extend T::Sig
+
+  def initialize(formula)
+    @formula = formula
+    @arguments = []
+    @working_dir = nil
+    @log_path = nil
+    @error_path = nil
+    @keep_alive = nil
+    @run_at_load = nil
+  end
+
+  def arguments(arguments = nil)
+    case arguments
+    when nil
+      @arguments
+    when Array
+      @arguments = arguments
+    else
+      raise TypeError, "Service#arguments expects an Array"
+    end
+  end
+
+  def working_dir(path = nil)
+    case path
+    when nil
+      @working_dir
+    when String
+      @working_dir = path
+    else
+      raise TypeError, "Service#working_dir expects a String"
+    end
+  end
+
+  def log_path(path = nil)
+    case path
+    when nil
+      @log_path
+    when String
+      @log_path = path
+    else
+      raise TypeError, "Service#log_path expects a String"
+    end
+  end
+
+  def error_path(path = nil)
+    case path
+    when nil
+      @error_path
+    when String
+      @error_path = path
+    else
+      raise TypeError, "Service#error_path expects a String"
+    end
+  end
+
+  def keep_alive(value = nil)
+    case value
+    when nil
+      @keep_alive
+    when TrueClass, FalseClass
+      @keep_alive = value
+    else
+      raise TypeError, "Service#keep_alive expects a Boolean"
+    end
+  end
+
+  def run_at_load(value = nil)
+    case value
+    when nil
+      @run_at_load
+    when TrueClass, FalseClass
+      @run_at_load = value
+    else
+      raise TypeError, "Service#run_at_load expects a Boolean"
+    end
+  end
+
+  # Returns a `Hash` of all instance variable values.
+  # @return [Hash]
+  def to_plist_hash
+    {
+      "Label"             => "homebrew.service.#{@formula.name}",
+      "KeepAlive"         => @keep_alive,
+      "RunAtLoad"         => @run_at_load,
```

Comment:
When is this changed and how?

Reply:
I don't understand the question

Reply:
Sorry, I mean: when is this set to true/false in existing plists and why? Will help figure out better names.

Reply:
> RunAtLoad: describes when the script should be run. There are a few different options here:

>    - RunAtLoad: run as soon as the job definition is loaded. Runs only once per load.
>    - StartInterval: start the job every n seconds. This example will run the job every 7200 seconds or every 2 hours. 
>   - StartCalendarInterval: run the job at a specific time and date. The below code will run the job every day at 9 AM.

Reply:
I'd be tempted to say we don't make this configurable at all. Thoughts?

Reply:
I made it configurable as `run_type`, but the other options aren't supported yet